### PR TITLE
Add PMD checks to project

### PR DESCRIPTION
**Pull request by etedpet**

### CODE REVIEW

**Comment by etedpet** [comment]
We can discuss if all these rules should be applied or if some of them should be excluded...

File: pom.xml [hunk 1]
```diff
@@ -82,6 +82,7 @@
         <org.eluder.coveralls.coveralls-maven-plugin.version>4.3.0</org.eluder.coveralls.coveralls-maven-plugin.version>
         <org.jacoco.jacoco-maven-plugin.version>0.8.2</org.jacoco.jacoco-maven-plugin.version>
         <org.pitest.pitest-maven.version>1.4.6</org.pitest.pitest-maven.version>
+        <org.apache.maven.plugins-maven-pmd-plugin.version>3.11.0</org.apache.maven.plugins-maven-pmd-plugin.version>
```
Comment (eperott) [inline pom.xml]: 3.12.0?

File: pom.xml [hunk 2]
```diff
@@ -298,6 +300,18 @@ limitations under the License.
...
                 </plugin>
+                <plugin>
+                    <artifactId>maven-pmd-plugin</artifactId>
+                    <version>${org.apache.maven.plugins-maven-pmd-plugin.version}</version>
+                    <configuration>
+                        <rulesets>
+                            <ruleset>${session.executionRootDirectory}/pmd-rules.xml</ruleset>
+                        </rulesets>
+                        <excludes>
+                            <exclude>com/ericsson/bss/cassandra/ecaudit/auth/cache/</exclude>
```
Comment (eperott) [inline pom.xml]: Let's remember to remove this exclude when we merge this into master (3.11) branch.

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java [hunk 3]
```diff
@@ -98,6 +95,7 @@
     private static final Set<Permission> DROP_PERMISSIONS = ImmutableSet.of(Permission.DROP);
     private static final Set<Permission> DESCRIBE_PERMISSIONS = ImmutableSet.of(Permission.DESCRIBE);
     private static final Set<Permission> AUTHORIZE_PERMISSIONS = ImmutableSet.of(Permission.AUTHORIZE);
+    private static final String UNEXPECTED_BATCH_STATEMENT = "Unexpected BatchStatement when mapping singe query for audit";
```
Comment (emolsson) [inline ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java]: Nit (was present before):

single

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/AuditPasswordAuthenticator.java [hunk 4]
```diff
@@ -104,13 +104,15 @@ public AuthenticatedUser legacyAuthenticate(Map<String, String> credentials) thr
...
+        private final AuditAdapter auditAdapter;
 
         private String decodedUsername;
 
-        AuditPlainTextSaslAuthenticator(InetAddress clientAddress, SaslNegotiator saslNegotiator)
+        AuditPlainTextSaslAuthenticator(InetAddress clientAddress, SaslNegotiator saslNegotiator, AuditAdapter auditAdapter)
         {
             this.clientAddress = clientAddress;
             this.saslNegotiator = saslNegotiator;
+            this.auditAdapter = auditAdapter;
```
[the remark this answers is not in the document]
Comment (etedpet) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/AuditPasswordAuthenticator.java]: Initially there where a warning that I "fixed" with this change:
https://pmd.github.io/pmd-6.8.0/pmd_rules_java_bestpractices.html#accessormethodgeneration

But there where a few of them and I didn't quite like the change so I made an exception to this rule... BUT i forgot to change back this code... Will revert this file change!

File: pom.xml (hunk 1 above)
Comment (etedpet) [reply pom.xml]: done

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/ResourceFactory.java [hunk 5]
```diff
@@ -26,16 +26,22 @@
...
+        // Utility class
+    }
+
     private static final String DATA_ROOT = "data";
     private static final String ROLES_ROOT = "roles";
     private static final String CONNECTIONS_ROOT = "connections";
     private static final String FUNCTIONS_ROOT = "functions";
 
     private static final String SEPARATOR = "/";
+    private static final String WORD_REGEX = "\\w+";
```
[the remark this answers is not in the document]
Comment (etedpet) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/ResourceFactory.java]: Done

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java (hunk 3 above)
Comment (etedpet) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java]: done

File: pom.xml (hunk 2 above)
Comment (etedpet) [reply pom.xml]: ok

**Pull request merged**